simtools/model/mirrors.py

```diff
@@ -54,26 +56,25 @@ def _read_mirror_list_from_ecsv(self):
         InvalidMirrorListFile
             If number of mirrors is 0.
         """
+        # Getting mirror parameters from mirror list.
 
-        # TODO - temporary hard wired geopars - should come from DB
-        self.diameter = 120
-        self.shape = 1
-        self._logger.debug(f"Shape = {self.shape}")
-        self._logger.debug(f"Diameter = {self.diameter}")
-
-        _mirror_table = Table.read(self._mirror_list_file, format="ascii.ecsv")
+        self.mirror_table = Table.read(self._mirror_list_file, format="ascii.ecsv")
         self._logger.debug(f"Reading mirror properties from {self._mirror_list_file}")
         try:
-            self._mirrors["flen"] = list(_mirror_table["mirror_panel_radius"].to("cm").value / 2.0)
-            self.number_of_mirrors = len(self._mirrors["flen"])
-            self._mirrors["number"] = list(range(self.number_of_mirrors))
-            self._mirrors["pos_x"] = [0.0] * self.number_of_mirrors
-            self._mirrors["pos_y"] = [0.0] * self.number_of_mirrors
-            self._mirrors["diameter"] = [self.diameter] * self.number_of_mirrors
-            self._mirrors["shape"] = [self.shape] * self.number_of_mirrors
+            self.shape_type = u.Quantity(self.mirror_table["shape_type"])[0].value
+            self.mirror_diameter = u.Quantity(self.mirror_table["mirror_diameter"])[0].value
+            self.number_of_mirrors = len(self.mirror_table["focal_length"])
+
+            self._logger.debug(f"Mirror shape_type = {self.shape_type}")
+            self._logger.debug(f"Mirror diameter = {self.mirror_diameter}")
+            self._logger.debug(f"Number of Mirrors = {self.number_of_mirrors}")
         except KeyError:
-            self._logger.debug(
-                f"Missing column for mirror panel focal length (flen) in {self._mirror_list_file}"
+            self._logger.debug("Mirror shape_type not in mirror file")
+            self.number_of_mirrors = len(self.mirror_table["mirror_id"])
```

Comment:
I wonder about this try..except. You might need to do it one by one because if the shape is not given but the diameter is, you won't read it. Also, I do not really understand why the number of mirrors is taken using a different key if the shape is no there.
Lastly, I thought we said that if the shape and diameter are not in the file, you will get them from the DB. That can be done in a separate PR though if you want.

Reply:
We could require them to be in the metadata. It might require some manual work for the person preparing the input file, but that's not too bad I guess. In that case, we must find a more friendly way to represent the shape then. The number 1, 2, 3 would not make sense to anyone external.